### PR TITLE
REST Integration tests improvements 2 replacement

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
REST API tests have been hard to decipher, modify and are sometimes not consistent.
These changes provide some scaffolding and changes to improve the integration tests on the REST side

- Update db query tests to use integration db ops method
- Updates spec files to contain their setup steps
- Moved domain setup logic from integrationDbOps.js into integrationDomainOps.js

**Which issue(s) this PR fixes**:
Partially addresses #412 with inspiration from #529
This replaces #594 

**Special notes for your reviewer**:
With this PR db and domain logic is centralized for the query and db integration tests.
There still remains some transfer list nuances that need to be brought over from #529 to ensure we're handling the various versions correctly

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by Nana-EC** [comment]
PR now has necessary r3 coverage with tests more streamlined.
This PR needs to go on so I can add the r4 logic of itemized non_fee_transfers that is in the importer side but not yet reflected in rest api. This will include both rest api product and test updates

File: hedera-mirror-rest/__tests__/specs/balances-01-no-args.spec.json [hunk 1]
```diff
@@ -32,10 +133,6 @@
         "account": "0.15.4",
         "balance": 40
       },
-      {
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/balances-01-no-args.spec.json]: 3 is the node account in the tests, so it's intentionally left out in the spec.

Also not true that it should have output as the balances query consults with the balance table. Without a balance set for an account it won't have a balance entry.

File: hedera-mirror-rest/__tests__/specs/transactions-02-specific-id.spec.json [hunk 2]
```diff
@@ -1,31 +1,79 @@
...
+  "setup": {
+    "accounts": [
+      {
+        "entity_num": 9
+      },
+      {
+        "entity_num": 10
+      }
+    ],
+    "balances": [
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/transactions-02-specific-id.spec.json]: You're  correct, we don't need it. Will remove

File: hedera-mirror-rest/__tests__/integrationDomainOps.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+const SERVICE_FEE = 4;
+
+let sqlConnection;
+let accountEntityIds;
+let recordFileId;
+
+const setUp = async function(testDataJson, sqlconn) {
+  accountEntityIds = {};
+  sqlConnection = sqlconn;
+  await addAccount(toAccount(TREASURY_ACCOUNT_ID));
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: I believe Mike's thinking here was to mimic the normal operations where a node and treasury account already exists before a new account is created.
Hence having that outside outside of the actual test logic and as part of the starting state of the database as you would see in production.

File: hedera-mirror-rest/__tests__/specs/transactions-none.spec.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/transactions-none.spec.json]: Updated

File: hedera-mirror-rest/__tests__/specs/transactions-r3.spec.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/transactions-r3.spec.json]: Updated

File: hedera-mirror-rest/__tests__/specs/balances-01-no-args.spec.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-rest/__tests__/specs/balances-01-no-args.spec.json]: I agree, it won't have output if there is no balance entry for it. I meant it should have a balance to match previous behavior.

File: hedera-mirror-rest/__tests__/integrationDomainOps.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: But treasury and nodes are just accounts themselves and shouldn't be treated differently. Also, what if we want to test a scenario where those accounts don't exist? Like your new no transactions test but for balances?

File: hedera-mirror-rest/__tests__/integrationDomainOps.js [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+  }
+};
+
+const addCryptoTransaction = async function(cryptoTransfer) {
+  if (!('senderAccountId' in cryptoTransfer)) {
+    cryptoTransfer.senderAccountId = cryptoTransfer.payerAccountId;
+  }
+  let sender = toAccount(cryptoTransfer.senderAccountId);
+  let recipient = toAccount(cryptoTransfer.recipientAccountId);
+  if (!('transfers' in cryptoTransfer)) {
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: Good note. Reverting to previous logic.

File: hedera-mirror-rest/__tests__/integrationDomainOps.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: Let me know if you feel strongly. I can set a balance for the node account or I can add a default account creation for the node with comments

[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: Gotcha will modify this.

[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: Moved the config into the spec. Makes sense to have state be more clear

File: hedera-mirror-rest/__tests__/integration.test.js [hunk 7]
```diff
@@ -203,35 +156,41 @@ function extractDurationAndMaxFeeFromTransactionResults(rows) {
...
+    return '@' + v['name'] + ',' + v['result'];
+  });
+}
+
 //
 // TESTS
 //
 
-test('DB integration test - transactions.reqToSql - no query string - 3 txn 9 xfer', async () => {
+test('DB integration test - transactions.reqToSql - no query string - 3 txn 21 xfers', async () => {
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/integration.test.js]: 9 xfers?

File: hedera-mirror-rest/__tests__/integration.test.js [hunk 8]
```diff
@@ -203,35 +156,41 @@ function extractDurationAndMaxFeeFromTransactionResults(rows) {
...
+    '@1051: account 98 \u01271',
     '@1052: account 8 \u0127-31',
-    '@1052: account 9 \u012730'
+    '@1052: account 9 \u012730',
+    '@1052: account 98 \u01271'
   ]);
 });
 
-test('DB integration test - transactions.reqToSql - single valid account - 1 txn 3 xfer', async () => {
+test('DB integration test - transactions.reqToSql - single valid account - 1 txn 7 xfers', async () => {
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/integration.test.js]: 3 xfers?

File: hedera-mirror-rest/__tests__/integration.test.js (hunk 7 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integration.test.js]: Fixed

File: hedera-mirror-rest/__tests__/integration.test.js (hunk 8 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integration.test.js]: Fixed

**Pull request merged**